pkg/kubelet/kubelet_node_status.go

```diff
@@ -404,18 +408,31 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 		return fmt.Errorf("nil %q node object", kl.nodeName)
 	}
 
+	podCIDRChanged := false
 	if node.Spec.PodCIDR != "" {
-		if err := kl.updatePodCIDR(node.Spec.PodCIDR); err != nil {
+		// Pod CIDR could have been updated before, so we cannot rely on
+		// node.Spec.PodCIDR being non-empty. We also need to know if pod CIDR is
+		// actually changed.
+		if podCIDRChanged, err = kl.updatePodCIDR(node.Spec.PodCIDR); err != nil {
 			glog.Errorf(err.Error())
 		}
 	}
 
 	kl.setNodeStatus(node)
+
+	now := kl.clock.Now()
+	if utilfeature.DefaultFeatureGate.Enabled(features.NodeLease) && now.Before(kl.lastStatusReportTime.Add(kl.nodeStatusReportFrequency)) {
+		if !podCIDRChanged && !nodeStatusHasChanged(&originalNode.Status, &node.Status) {
```

Comment:
`tryUpdateNodeStatus` does not touch label. I don't think we need to check labels here. Also confirmed with @yujuhong offline.

Reply:
@verult where was the CSI label-updating going to hook in? I thought it folded into the node status update loop

Reply:
As far as I can tell, label updates are separate from status updates. There are 2 functions touching labels, `initialNode()` and `updateDefaultLabels()`. And both results are sent to master in `tryRegisterWithAPIServer()` [here](https://github.com/kubernetes/kubernetes/blob/8dcdec0a6700b04beb9f73b66f5b125297c9acc7/pkg/kubelet/kubelet_node_status.go#L121).

Reply:
Thanks for your reply, @verult! I see what it means now. IMO, it is fine to leave the CSI label-updating as is because:
1. It is using `PatchNodeStatus` to update status/labels/annotations already, which is separate from kubelet's node status update.
2. It is already checking if update is needed. That means, it only posts node status to master when there is some change, which is consistent with what we want to achieve with node lease feature.

WDYT, @liggitt?

Reply:
Kubelet itself updates the node labels in the registration function, which is separate from the node status update loop.
https://github.com/kubernetes/kubernetes/blob/ed06cbe3e32370da07e9035f89a28521372c9392/pkg/kubelet/kubelet_node_status.go#L117-L125

As it stands, kubelet is not using this loop to update any node labels (unless I missed something). I don't think it's necessary to check the labels unless someone intends to use it.